api/authenticator.py

```diff
@@ -2254,6 +2286,11 @@ def _authentication_flow_document(self, _db):
             links=[dict(rel="authenticate",
                         href=self._internal_authenticate_url(_db))]
         )
+        if self.LOGIN_BUTTON_IMAGE:
+            # TODO: I'm not sure if logo is appropriate for this, since it's a button
+            # with the logo on it rather than a plain logo. Perhaps we should use plain
+            # logos instead.
+            flow_doc["links"] += [dict(rel="logo", href=url_for("static_image", filename=self.LOGIN_BUTTON_IMAGE, _external=True))]
```

Comment:
What's the difference between this one and the one above?

Reply:
They're the same but the basic and oauth auth providers have different implementations of this function so I put it in two places.